exporter/signalfxexporter/internal/translation/translator.go

```diff
@@ -671,8 +671,9 @@ func (mp *MetricTranslator) translateDimension(orig string) string {
 	return orig
 }
 
-// aggregateDatapoints aggregates datapoints assuming that they have
-// the same Timestamp, MetricType, Metric and Source fields.
+// aggregateDatapoints makes sure all Metrics with matching Timestamp, MetricType, Metric and Source fields
+// are aggregated.
+// NOTE: Assumes all input datapoints share metric name.
 func aggregateDatapoints(
```

Comment:
Actually if we move `sfxDataPoints = c.translateAndFilter(sfxDataPoints)` to the scope of one resource as I suggested above, I don't think this pre-aggregation is needed at all. It'll be the same input for `aggregateDatapoints` as before because each `pmetric.ResourceMetrics` object has only one `pmetric.Metric` of a particular name

Reply:
I did some investigation, and this is what I found. The name check is required in the `case ActionAggregateMetric` to make sure the rule that is being checked actually applies to each metric. However, with the collector running a `hostmetrics` receiver, this check never fails: 
```
// Datapoints match if they have same name, type, time, and source.
if metricTypesMatch &&
    dpsToAggregate[0].Timestamp == dps[j].Timestamp &&
    dpsToAggregate[0].Source == dps[j].Source {
```

So theoretically you're right, we might be able to remove this entire method. I think it's a fair assumption that if metrics have the same name they'll also have the same type. However, I'm more worried about the `timestamp` and `source` attributes.

Is it always guaranteed that datapoints within the same metric will have identical sources and timestamps?

Reply:
> Is it always guaranteed that datapoints within the same metric will have identical sources and timestamps?

Yes, scraping receivers are always set the same timestamp on one resource metric. I'm not sure what `Source` is. Do we even set this field in signalfx exporter?

Reply:
I'm actually not seeing the `Source` field set in the exporter. I would have expected it to be set when converting from pmetric format to signalfx proto format, but it's never referenced. From this, I think it's safe to remove this whole method. I'll test again as well.